src/app/components/queue/form/QuestionForm.tsx

```diff
@@ -0,0 +1,209 @@
+      // TODO(lnguye2693) - Display error
+    }
+
+    setOpenForm(false);
+  };
+  return (
+    <Box>
+      {trigger}
+      <Drawer open={openForm} anchor="bottom">
+        <Box height="100vh" width="100vw" overflow="scroll">
+          <Box display="flex" alignItems="center" padding={1.5} gap={1}>
+            <button
+              onClick={() => {
+                setOpenForm(false);
+              }}
+              style={{
+                background: "transparent",
+                border: "none",
+                display: "flex",
+                justifyContent: "center",
+                alignItems: "center",
+                aspectRatio: 1,
+              }}
+            >
+              <IoMdClose size={20} />
+            </button>
+            <Box>{title}</Box>
+          </Box>
+          <Box display="flex" flexDirection="column" gap={2} padding={2}>
+            <TextField
+              required
+              label="Question"
+              placeholder="Give your question a title"
+              focused
+              value={question}
+              onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                setQuestion(event.target.value);
+              }}
+            ></TextField>
+
+            <TextField
+              required
+              label="Description"
+              placeholder="Add description"
+              focused
+              multiline
+              rows={4}
+              value={description}
+              onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                setDescription(event.target.value);
+              }}
+            ></TextField>
+
+            {/* TODO(lnguyen2693) - Add index for tags, sort and display them 
+            in an order */}
+            {Object.keys(ohContext.course.tags).map((k) =>
+              ohContext.course.tags[k].multipleChoice ? (
+                <MultipleChoiceTags
+                  key={k}
+                  tagsKey={k}
+                  tags={ohContext.course.tags[k]}
+                  allQuestionTags={questionTags}
+                  updateQuestionTags={updateQuestionTags}
+                ></MultipleChoiceTags>
+              ) : (
+                <SingleChoiceTags
+                  key={k}
+                  tagsKey={k}
+                  tags={ohContext.course.tags[k]}
+                  allQuestionTags={questionTags}
+                  updateQuestionTags={updateQuestionTags}
+                ></SingleChoiceTags>
+              )
+            )}
+
+            <FormControl>
+              <FormLabel>Optional</FormLabel>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
+                      setQuestionPublic(event.target.checked)
+                    }
+                    checked={questionPublic}
+                  />
+                }
+                label={
+                  <Box>
```

Comment:
I think this can use some more spacing.

<img width="387" alt="image" src="https://github.com/user-attachments/assets/8c8b4a1b-690e-4db8-8b38-014f0c9e529e">